nevergrad/functions/corefuncs.py

```diff
@@ -3,53 +3,67 @@
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
 
-import typing as tp
 from math import exp, sqrt, tanh
 import numpy as np
-from nevergrad.parametrization import discretization
+import nevergrad.common.typing as tp
 from nevergrad.common.decorators import Registry
 
 
 registry: Registry[tp.Callable[[np.ndarray], float]] = Registry()
 
 
-def onemax(x: tp.List[int]) -> float:
-    """onemax(x) is the most classical case of discrete functions, adapted to minimization.
-
-    It is originally designed for lists of bits. It just counts the number of 1,
-    and returns len(x) - number of ones..
-    It also works in the continuous case but in that cases discretizes the
-    input domain by ]0.5,1.5] --> 1 and 0 everywhere else.
-    """
-    return len(x) - sum(1 if int(round(w)) == 1 else 0 for w in x)
-
-
-def leadingones(x: tp.List[int]) -> float:
-    """leadingones is the second most classical discrete function, adapted for minimization.
-
-    Returns len(x) - number of initial 1. I.e.
-    leadingones([0 1 1 1]) = 4,
-    leadingones([1 1 1 1]) = 0,
-    leadingones([1 0 0 0]) = 3.
-    """
-    for i, x_ in enumerate(list(x)):
-        if int(round(x_)) != 1:
-            return len(x) - i
-    return 0
-
-
-def jump(x: tp.List[int]) -> float:  # TODO: docstring?
-    """There exists variants of jump functions; we are in minimization.
-
-    The principle of a jump function is that local descent does not succeed.
-    Jumps are necessary.
-    """
-    n = len(x)
-    m = n // 4
-    o = n - onemax(x)
-    if o == n or o <= n - m:
-        return n - m - o
-    return o  # Deceptive part.
+class DiscreteFunction:
+
+    def __init__(self, name: str, arity: int = 2) -> None:
+        """onemax(x) is the most classical case of discrete functions, adapted to minimization.
+
+        It is originally designed for lists of bits. It just counts the number of 1,
+        and returns len(x) - number of ones..
+        It also works in the continuous case but in that cases discretizes the
+        input domain by ]0.5,1.5] --> 1 and 0 everywhere else.
+        leadingones is the second most classical discrete function, adapted for minimization.
+
+        Returns len(x) - number of initial 1. I.e.
+        leadingones([0 1 1 1]) = 4,
+        leadingones([1 1 1 1]) = 0,
+        leadingones([1 0 0 0]) = 3.
+        There exists variants of jump functions; we are in minimization.
+
+        The principle of a jump function is that local descent does not succeed.
+        Jumps are necessary.
+        """  # TODO update to explain arities and other than leadingones + the examples are actually wrong
```

Comment:
This description must be updated, the examples do not match the actual functions

Reply:
Done.

Reply:
I still don't see how that is pseudo random, that seems very deterministic to me :s

Reply:
Pseudo-random means deterministic actually. This is just a very naive pseudo-random.